core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -234,9 +236,14 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val actorSyste
         properties += ((key, value))
       }
     }
+
     // TODO (prashant) send conf instead of properties
     driverActor = actorSystem.actorOf(
       Props(new DriverActor(properties)), name = CoarseGrainedSchedulerBackend.ACTOR_NAME)
+
+    // If a principal and keytab have been set, use that to create new credentials for executors
+    // periodically
+    SparkHadoopUtil.get.scheduleLoginFromKeytab()
```

Comment:
All the logic for this is really in the YarnSparkHadoopUtil and it only applies to yarn.  I could see some of this possibly being reused for standalone mode but everything else is setup for just yarn, so perhaps a better place for this is in the YarnSchedulerBackend.   Which brings me to my second thought which is we seem to be putting a lot of stuff in YarnSparkHadoopUtil now.  It used to be just utility functions and now we have threads and stuff not interfaces through SparkHadoopUtil.    Perhaps having some sort of security/token manager would be a better split.

thoughts from others on this?